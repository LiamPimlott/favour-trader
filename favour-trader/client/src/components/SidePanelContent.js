import React, { Component } from 'react';
import axios from 'axios'
import './SidePanelContent.css';
import { Link } from 'react-router-dom';
import {Menu, Icon} from 'antd';

const SubMenu = Menu.SubMenu;

class SidePanelContent extends Component {
    constructor() {
        super();
        this.state = {
            allSkills: [],
            hasSkills: [],
            wantSkills: [],
            openKeys: [],
        };
        this.rootSubmenuKeys = ['sub1', 'sub2', 'sub4'];
    }

    onOpenChange = (openKeys) => {
        const latestOpenKey = openKeys.find(key => this.state.openKeys.indexOf(key) === -1);
        if (this.rootSubmenuKeys.indexOf(latestOpenKey) === -1) {
            this.setState({ openKeys });
        } else {
            this.setState({
                openKeys: latestOpenKey ? [latestOpenKey] : [],
            });
        }
    };

    componentDidMount() {
        const { authService } = this.props;
        if (authService.loggedIn()) {
            const userProfile = authService.getProfile();
            const config = {
               headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
                'Authorization': authService.getToken(),
                }
            };

            axios.get('/api/skills/all')
            .then(res => res.data)
            .then(data => this.setState({allSkills: data }));

            if (!!userProfile && userProfile.email) {
                const userEmail = userProfile.email;
                axios.post('/api/users/has', {
                    email: userEmail,
                }, config)
                .then(res => res.data.user)
                .then(user => this.setState({ hasSkills: user.has }));

                axios.post('/api/users/wants', {
                    email: userEmail,
                }, config)
                .then(res => res.data.user)
                .then(user => this.setState({ wantSkills: user.wants }));
            }
        }
    }

    renderSkills(skillSet) {
        const skills = this.state[skillSet];
        return (
            <div>
                {
                    (skills !== []) ? (
                        <ul>
                            {
                                skills.map(function (skill) {
                                    return (<li key={skill._id}> {skill.skill} </li>)
                                }, this)
                            }
                        </ul>
                    ) : (
                            ''
                    )
                }
            </div>
        );
    }

    render() {
        return (
                <Menu theme="light"
                      mode="inline"
                      openKeys={this.state.openKeys}
                      onOpenChange={this.onOpenChange}
                      style={{ height: '100%'}}>
                    <Menu.Item key="sub1">
                        <Link to="/Profile">
                            <Icon type="user"/> <span>My Profile</span>
                        </Link>
                    </Menu.Item>
                    <SubMenu key="sub2" title={<span><Icon type="search" /><span>Find Matches</span></span>}>
						<Menu.Item key="4">
							<Link to="/">
								Perfect Matches
							</Link>
						</Menu.Item>
                        <Menu.Item key="5"
							onClick={ ()=>{this.props.params.filtertype} }>
							<Link to="/filterHas">
								What I Have
							</Link>
						</Menu.Item>
                        <Menu.Item key="6">
							<Link to="/filterWants">
								What I Want
							</Link>
						</Menu.Item>
                    </SubMenu>
                    <SubMenu key="sub3" title={<span><Icon type="book" /><span>Trades</span></span>}>
                        <Menu.Item key="5">Active</Menu.Item>
                        <Menu.Item key="6">Offers</Menu.Item>
                    </SubMenu>
                </Menu>
        );
    }
}

export default SidePanelContent;