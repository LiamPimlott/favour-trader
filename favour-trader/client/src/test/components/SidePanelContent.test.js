import React from 'react';
import renderer from 'react-test-renderer';
import SidePanelContent from '../../components/SidePanelContent.js';
import AuthServiceMock from './AuthServiceMock.js';
import {MemoryRouter} from 'react-router-dom';

it('renders correctly', () => {
    var authServiceMock = new AuthServiceMock();
    const tree = renderer
        .create(<MemoryRouter>
            <SidePanelContent authService={authServiceMock} />
        </MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});