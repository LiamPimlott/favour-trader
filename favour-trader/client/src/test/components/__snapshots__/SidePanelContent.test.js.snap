// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  aria-activedescendant=""
  className="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
  onKeyDown={[Function]}
  role="menu"
  style={
    Object {
      "height": "100%",
    }
  }
  tabIndex="0"
>
  <li
    aria-disabled={undefined}
    aria-selected={false}
    className="ant-menu-item"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={
      Object {
        "paddingLeft": 24,
      }
    }
    title={undefined}
  >
    <a
      href="/Profile"
      onClick={[Function]}
    >
      <i
        className="anticon anticon-user"
      />
       
      <span>
        My Profile
      </span>
    </a>
  </li>
  <li
    className="ant-menu-submenu ant-menu-submenu-inline"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={undefined}
  >
    <div
      aria-expanded={false}
      aria-haspopup="true"
      aria-owns="sub2$Menu"
      className="ant-menu-submenu-title"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "paddingLeft": 24,
        }
      }
      title={undefined}
    >
      <span>
        <i
          className="anticon anticon-search"
        />
        <span>
          Find Matches
        </span>
      </span>
      <i
        className="ant-menu-submenu-arrow"
      />
    </div>
  </li>
  <li
    className="ant-menu-submenu ant-menu-submenu-inline"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={undefined}
  >
    <div
      aria-expanded={false}
      aria-haspopup="true"
      aria-owns="sub3$Menu"
      className="ant-menu-submenu-title"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "paddingLeft": 24,
        }
      }
      title={undefined}
    >
      <span>
        <i
          className="anticon anticon-book"
        />
        <span>
          Trades
        </span>
      </span>
      <i
        className="ant-menu-submenu-arrow"
      />
    </div>
  </li>
</ul>
`;
